catalyst/src/enode.js

```diff
@@ -0,0 +1,265 @@
+
+// dom manipulation package
+function toNodeValue(sel, context){
+  context = context || document;
+  if (!sel){
+    return [];
+  } else if (typeof sel === 'string'){
+    if (sel[0] === '<') {
+      var div = context.createElement('div');
+      div.innerHTML = sel.trim();
+      return [div.firstChild]; 
+    } else {
+      return context.querySelectorAll(sel);
+    }
+  } else if (sel instanceof Element){
+    return [sel]
+  } else if (sel.constructor === ENode){
+    return sel.el
+  } else if (Array.isArray(sel)){
+    return sel
+  } else {
+    return [];
+  }
+}
+
+export const ENode = function(sel, context){
+  context = context || document;
+  var el = toNodeValue(sel, context);
+  this.el = Array.prototype.slice.call(el);
+  this.length = this.el.length;
+};
+
+//fltering
+ENode.prototype.filter = function(sel){
+  var el = this.el;
+  if (!sel) return this;
+  return new ENode(el.filter(function(e){
+    return e.matches(sel)
+  }))
+};
+ENode.prototype.contains = function (text) {
+  var el = this.el;
+  return new ENode(el.filter(function (e) { return e.textContent.includes(text)}));
```

Comment:
I would change `return e.textContent.includes(text)` to `return e.textContent.indexOf(text) > -1` as IE11 doesn't support `includes`.  See https://caniuse.com/?search=includes.  Even though we exclude IE from our experiments, the context still runs for VBG and they often complain of errors they see on IE11.